### PR TITLE
Updating Changelog for 0.2.1-patch3

### PR DESCRIPTION
**Pull request by jaiveerk**
# Description

Updating changelog for latest patch updating externaldns version

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration. Is it a breaking change which will impact consuming tool(s)?

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by OliverMKing — APPROVED** [review]

**Review by bfoley13 — APPROVED** [review]

File: .github/workflows/e2e.yaml [hunk 1]
```diff
@@ -66,8 +66,10 @@ jobs:
       with:
         ref: ${{ inputs.checkout_ref }}
     
-    - if: ${{ !(needs.test.status == 'success')}}
-      run: exit 1 # will make status show as failure
+    - if: ${{ !(needs.test.conclusion == 'success')}}
+      run: |
+        echo "job status: ${{ needs.test.conclusion }}"
```
Comment (OliverMKing) [inline .github/workflows/e2e.yaml:71]: there's something bugged about the existing logic but this isn't the fix. revert and we will fix in another pr.

this isn't the correct syntax for doing this. You'd use an environment variable to get the github variable passed to the bash

**Pull request merged**